internal/lsp/server.go

```diff
@@ -91,6 +147,12 @@ type Server struct {
 	watchers       core.Set[project.WatcherHandle]
 	logger         *project.Logger
 	projectService *project.Service
+
+	// enables tests to share a cache of parsed source files
+	getCachedSourceFile func(string, tspath.Path, core.ScriptTarget) *ast.SourceFile
```

Comment:
I'm wary of this but I'm not sure what to do otherwise. The trouble is that this would bypass all of the logic the server/service/project that do their own caching, but perhaps we just need to test that without fourslash.